### DISCUSSION TITLE
toMatchFileSnapshot always mismatched if empty file (CI only)

### DISCUSSION THREAD

**Discussion by gterras** [vitest-dev/vitest#5890] [Q&A] [answered]
Hi, 

I've spent hours trying to pinpoint what could cause this, the following works as expected on any machine but will systematically fail in Gitlab CI (docker):

```ts
// only fails in CI and if referenced file is empty
// cannot be reproduced locally 
context.expect('').toMatchFileSnapshot(EMPTY_FILE)
```

> Error: Snapshot `xxx` mismatched

Without any further info (even "expected/received" labels aren't displayed). 

I've tried `null` and `undefined` with same result. It's like for some reason `toMatchFileSnapshot` returns something not matchable (error?), but **only in CI and when the file is empty (every other case work as expected)**. Same result whatever the node version (18, 20, 22), I've tried different linux image with no success, cannot reproduce locally. It works as expected if the file contains at least one character. 

The referenced file was originally written by vitest. 

I'm really running out of things to try. Thanks for help!

**Comment by hi-ogawa** [comment]
Can you try running locally with `CI=true vitest ...`?
Vitest changes some default behavior based on `CI` env var, so you might be able to reproduce it locally.

**Reply by gterras** [reply]
I confirm adding `CI=true` triggers the same behavior locally. Do you have any idea of what can cause this, or a way to debug it? 

edit: running `CI=false vitest ...` in CI doesn't solve the problem unfortunately

**Reply by hi-ogawa** [reply, marked as answer]
Thanks for confirming. Can you submit a issue with a repro?
Vitest probably got confused by empty snapshot file and tries to create a new snapshot, but that's not allowed on CI environment (i.e. it's treated as error by default).

https://github.com/vitest-dev/vitest/blob/1dfcafcc659ffc774b8512cbbdfdaf590d5b5ef8/packages/vitest/src/node/config.ts#L270-L274

**Reply by gterras** [reply]
Thanks, will do this tomorrow.

**Reply by gterras** [reply]
FYI https://github.com/vitest-dev/vitest/issues/5891
